### PR TITLE
Passed Deaf Test and other failings; needs time to rip

### PR DESCRIPTION
**Pull request by eburdon**

### CODE REVIEW

File: src/paxos/paxos.go [hunk 1]
```diff
@@ -158,14 +149,19 @@ func (px *Paxos) Decide(args *Proposal, reply *DecideReply) error {
 	defer px.mu.Unlock()
 
 	if _, ok := px.recProposals[args.Seq]; !ok {
-		return nil
+		// return a new empty proposal
+		px.recProposals[args.Seq] = new(Proposal)
+		px.recProposals[args.Seq].Fate = Pending
+		px.recProposals[args.Seq].Seq = 0
+		px.recProposals[args.Seq].PropNum = 0
+		px.recProposals[args.Seq].Value = ""
```
Comment (eburdon) [inline src/paxos/paxos.go:157]: Not sure if this actually improved things; added to make explicit

File: src/paxos/paxos.go [hunk 2]
```diff
@@ -434,12 +487,15 @@ func (px *Paxos) StartProtocol(seq int, v interface{}) {
...
-
 		propose_ok, N, V := px.HandlePropose(proposal)
 
 		if propose_ok {
+			// fmt.Println("\n\n\nCalling handle accept with...")
+			// fmt.Println(N, V)
+			// fmt.Println("\n\n\n")
+
+			// ignore the proposal V, Seq?
+			proposal.Value = V
```
Comment (eburdon) [inline src/paxos/paxos.go:498]: Passing the propose_ok value forward

File: src/paxos/paxos.go [hunk 3]
```diff
@@ -331,48 +353,82 @@ func (px *Paxos) Propose(args *Proposal, reply *InterrogationReply) error {
...
+					// set reply
+					// if it doesn't equal my proposed value, I know it
+					// must have been DECIDED previously
+					// what if I have two values, from two servers with
+					// the same sequence? Tiebreaker == highest PropNum
+					if peer_reply.Value != proposal.Value {
+						if peer_reply.Value != nil {
+							replySeq = peer_reply.Seq
+							replyValue = peer_reply.Value
+						}
```
Comment (eburdon) [inline src/paxos/paxos.go:429]: This here was the major change. Retrieving the K, V (interrogation) from the peers. I feel this is brittle, but wasn't sure how else to tackle it.

File: src/paxos/paxos.go (hunk 2 above)
Comment (jordan-heemskerk) [inline src/paxos/paxos.go:498]: :+1: I found this one too :smile:

File: src/paxos/paxos.go (hunk 3 above)
Comment (jordan-heemskerk) [inline src/paxos/paxos.go:429]: :+1:

**Pull request merged**